tests/application_development/code_relocation/src/main.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <ztest.h>
+
+/* This function will allow execute from sram region.
+ * This is needed only for this sample because by default all soc will
+ * disable the execute from SRAM.
+ * An application that requires that the code be executed from SRAM will
+ *  have to configure the region appropriately in arm_mpu_regions.c.
+ */
+
+#if (defined(CONFIG_ARM_MPU) && !defined(CONFIG_CPU_HAS_NXP_MPU))
+#include <zephyr/arch/arm/aarch32/cortex_m/cmsis.h>
+void disable_mpu_rasr_xn(void)
+{
+	uint32_t index;
+	/* Kept the max index as 8(irrespective of soc) because the sram
+	 * would most likely be set at index 2.
+	 */
+	for (index = 0U; index < 8; index++) {
```

Comment:
get_num_regions is defined internally, and I can not call directly., and CONFIG_NUM_MPU_REGIONS is not always defined

Reply:
Is the number of the MPU regions in all the platforms that allowed for this test always 8? My original option is using `MPU_TYPE` when CONFIG_NUM_MPU_REGIONS is not defined. However, similarly, I won't persist since it is only for cortex-M